Source/Alamofire.swift

```diff
@@ -116,303 +116,309 @@ extension URLRequest {
-/// - parameter encodingCompletion:      The closure called when the `MultipartFormData` encoding is complete.
-public func upload(
-    multipartFormData: @escaping (MultipartFormData) -> Void,
-    usingThreshold encodingMemoryThreshold: UInt64 = MultipartUpload.multipartFormDataEncodingMemoryThreshold,
-    to url: URLConvertible,
-    method: HTTPMethod = .post,
-    headers: HTTPHeaders? = nil) -> UploadRequest
-{
-    return SessionManager.default.upload(
-        multipartFormData: multipartFormData,
-        usingThreshold: encodingMemoryThreshold,
-        to: url,
-        method: method,
-        headers: headers
-    )
-}
-
-/// Encodes `multipartFormData` using `encodingMemoryThreshold` and the default `SessionManager` and
-/// calls `encodingCompletion` with new `UploadRequest` using the `urlRequest`.
-///
-/// It is important to understand the memory implications of uploading `MultipartFormData`. If the cummulative
-/// payload is small, encoding the data in-memory and directly uploading to a server is the by far the most
-/// efficient approach. However, if the payload is too large, encoding the data in-memory could cause your app to
-/// be terminated. Larger payloads must first be written to disk using input and output streams to keep the memory
-/// footprint low, then the data can be uploaded as a stream from the resulting file. Streaming from disk MUST be
-/// used for larger payloads such as video content.
-///
-/// The `encodingMemoryThreshold` parameter allows Alamofire to automatically determine whether to encode in-memory
-/// or stream from disk. If the content length of the `MultipartFormData` is below the `encodingMemoryThreshold`,
-/// encoding takes place in-memory. If the content length exceeds the threshold, the data is streamed to disk
-/// during the encoding process. Then the result is uploaded as data or as a stream depending on which encoding
-/// technique was used.
-///
-/// - parameter multipartFormData:       The closure used to append body parts to the `MultipartFormData`.
-/// - parameter encodingMemoryThreshold: The encoding memory threshold in bytes.
-///                                      `multipartFormDataEncodingMemoryThreshold` by default.
-/// - parameter urlRequest:              The URL request.
-/// - parameter encodingCompletion:      The closure called when the `MultipartFormData` encoding is complete.
-public func upload(
-    multipartFormData: @escaping (MultipartFormData) -> Void,
-    usingThreshold encodingMemoryThreshold: UInt64 = MultipartUpload.multipartFormDataEncodingMemoryThreshold,
-    with urlRequest: URLRequestConvertible) -> UploadRequest
-{
-    return SessionManager.default.upload(
-        multipartFormData: multipartFormData,
-        usingThreshold: encodingMemoryThreshold,
-        with: urlRequest
-    )
+/// Global namespace containing API for the `default` `SessionManager` instance.
+public enum Alamofire {
```

Comment:
Hi, @jshier ,  there is a question - why do you choice `enum` rather than `struct` here？I wonder to know the difference between them. Thank you.

Reply:
By convention, since Swift doesn’t have namespaces, enums are used instead, as they can’t be accidentally instantiated.

Reply:
got it. Thanks for your reply~👍